client/components/router/guestOrRedirect.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import {connect} from 'react-redux'
+import {push} from 'react-router-redux'
+
+import selectors from '../../store/selectors'
+import userClientRole from '../../lib/user-client-role'
+
+const mapStateToProps = state => ({
+  user: selectors.auth.getUser(state)
+})
+
+const mapDispatchToProps = dispatch => ({
+  push: location => dispatch(push(location))
+})
+
+const redirectIfAlreadySignedIn = props => {
+  if (props.user && props.user._id) {
+    const role = userClientRole(props.user)
+    props.push(role ? `/${role.split('/')[1]}s` : '/')
+  }
+}
+
+const guestOrRedirect = WrappedComponent => {
+  class GuestOrRedirect extends React.Component {
+    constructor(props) {
+      super(props)
+      redirectIfAlreadySignedIn(props)
+    }
+  
+    componentWillReceiveProps(nextProps) {
+      if (this.props.user !== nextProps.user)
```

Comment:
Won't this be true if the user updates their profile too?

Reply:
It was an attempt to limit the number of times `redirectIfAlreadySignedIn()` was called, but now that I think about it, Redux `connect()` is already doing this. I'll remove it.

It would be true, but I don't see that being a problem. If the component hadn't redirected a user away from the sign in page at that point, the `redirectIfAlreadySignedIn()` should be called again.

Reply:
Oh yeah it's only used for pages that shouldn't be accessible by logged in users, ignore me!